core/blockchain.go

```diff
@@ -2278,14 +2278,14 @@ func (bc *BlockChain) reorg(oldHead *types.Header, newHead *types.Block) error {
 	// as it will be handled separately outside of this function
 	for i := len(newChain) - 1; i >= 1; i-- {
 		// Insert the block in the canonical way, re-writing history
-		bc.writeHeadBlock(newChain[i])
+		newBlock = bc.GetBlock(newChain[i].Hash(), newChain[i].Number.Uint64())
```

Comment:
This begins at the latest in the new chain. 

Q1: Is the latest block in the new chain already retrievable via `bc.GetBlock`? I think the answer is Yes. 

Q2: Isn't it the same as the incoming parameter `newHead`? If so, we can use that. Most calls to reorg will only be 1 block, right?